source/microbit/microbitmusic.cpp

```diff
@@ -268,21 +281,11 @@ STATIC mp_obj_t microbit_music_get_tempo(void) {
 }
 MP_DEFINE_CONST_FUN_OBJ_0(microbit_music_get_tempo_obj, microbit_music_get_tempo);
 
-STATIC mp_obj_t microbit_music_stop(mp_uint_t n_args, const mp_obj_t *args) {
-    const microbit_pin_obj_t *pin;
-    if (n_args == 0) {
-        pin = &microbit_p0_obj;
-    } else {
-        pin = microbit_obj_get_pin(args[0]);
-    }
-    pwm_set_duty_cycle(pin->name, 0);
-    microbit_obj_pin_free(pin);
-
-    async_music_state = ASYNC_MUSIC_STATE_IDLE;
-
+STATIC mp_obj_t microbit_music_stop(void) {
+    music_stop();
     return mp_const_none;
 }
-MP_DEFINE_CONST_FUN_OBJ_VAR_BETWEEN(microbit_music_stop_obj, 0, 1, microbit_music_stop);
+MP_DEFINE_CONST_FUN_OBJ_0(microbit_music_stop_obj, microbit_music_stop);
```

Comment:
Thanks for your PR.

I agree with @dpgeorge's request. We must not change the API or we'll risk annoying lots of people whose code suddenly won't work. ;-)

Please revise! Thank you! :-)